### ISSUE TITLE
Loader factory is getting "lost"

### ISSUE THREAD

**Issue by kptLucek** [angular-translate/angular-translate#1742]
### Loader factory is getting "lost"
I've got an application that translates herself. Basically it works like that:
- add language to application
- fill translations for translation keys for given language
- reload translations from backend

I'm also using `ui.router`, and using `languageCode` in my abstract state to load & use requested language:
```ts
//...
{
        name:     'app.skeleton',
        url:      '/{language:(?:(cs|en|sk|de)}',
        abstract: true,
        params:   {
            language: {
                value:   'en',
                dynamic: true
            },
        },
        resolve:  {
            trans: [
                '$translate',
                '$stateParams',
                function (
                    $translate: angular.translate.ITranslateService,
                    $stateParams: angular.ui.IStateParamsService
                ) {
                    $translate.use($stateParams['language']);
                }
            ]
        }
    }
//...
```

I'm using custom `TranslationLoaderFafctory`:
```ts
// ...
function factoryBody(
            StageRespository: StageRepository,
            FetcherService: FetcherService,
            $log: angular.ILogService
        ) {
            let stage = StageRespository.getStage();
            return function (options: { key: string }) {
                let path = stage.apiUrl + '/' + stage.applicationContext + '/translations';
                $log.error('OPTIONS: ', options);
                return FetcherService
                    .raw(path, 'get', {language: options.key})
                    .then((response) => {
                        return response;
                    }, (error) => {
                        $log.error(error);
                    });
            }
        }
// ....
```

```ts
// language change logic
swapLanguage = (language: ILanguage) => {
        //...
        this.$translate.use(language.name).then(() => {
            this.$translate.refresh(language.name);
        });
        //...
    };
```
```ts
//submit of translation update & reloading translations
public submit = () => {
        //...
        this.$postman.createRequest(path, method, this.translation).then((response: ITranslation) => {
            //...
            return response;
        }).then((response: ITranslation) => {
            let usedLanguageCode = this.language.name;
            if (this.$translate.use() === usedLanguageCode) {
                // only refresh, because $translate is already using our current language
                this.$translate.refresh(usedLanguageCode);
            }
            return response;
        });
    }
```

### ENV
* version of angular-translate: 2.14.0
* version of angular: 1.6.1
* which browser and its version: Chromium, 58.0

### Steps to reproduce
```Will add once i'll get back home - need to reproduce in raw js```

### Expected behaviour
Tell us what should happen
As translator for given application, i should be able to translate given `translation_key` in given `language`, after updating given translation the `$translator` is getting informed (if currentLang === `.use()`) about change in translation. They're getting reloaded and displayed as expected, but then, on the same page i want to translate different `language` in the same `translation_key` so `$translator` won't be informed in this case and won't reload translations. After changing whole application language (lets say from `en` to `cs`) then translator is loading translations for `cs` language, replacing `translation_keys` used. 
### Actual behaviour
Tell us what happens instead
Lets say that:
i'm on `en` language right now (whole application, and `$translator` is informed about using that language) and i'm translating both, `en` and `cs` language as follows:
1. do some translation on `translation_key` using `en` language
2. save
3. translator reloads translations (same "context")
4. change translated language to `cs` ("context" is `en`)
5. do some translations
6. save
7. change context (inform app & `$translator`) to use `cs`
8. repeat steps 1-7 for given language (`cs`)
9. switch back to `en`

After step 9. `$translator` forgot about my `TranslationLoaderFafctory`, translations are no longer being reloaded, no error provided

**Comment by knalli** [comment]
Hello,

as we had already mentioned in the [issue's template](https://github.com/angular-translate/angular-translate/blob/master/.github/ISSUE_TEMPLATE.md), we need a working and as minimal as possible demo covering your specific use case or issue. 

Follow these steps:
1. Please open [this plnkr base](https://plnkr.co/edit/8vRLnfGmtg51qZFfDeVc?p=preview).
2. Fork it (at the top left).
3. Ensure the version of AngularJS and angular-translate is correct.
4. Ensure all angular-translate plugins are available; reduce and remove anything you can.
5. Write a minimal as possible demo for your specific issue.
6. Save and freeze. Provide us the final link to your demo.

**Comment by knalli** [comment]
Hello,

this is a general reminder: We are still waiting for your reply with additional details. If you cannot help us, we cannot help you.

Looking forward to hear from you.

**Comment by knalli** [comment]
Hello,

I'm closing this issue due inactivity. We have asked for additional details, but you could not helped us. Therefore we close this issue as not reproducible.

If you still have an issue, please create a new issue because we cannot be sure to track closed issues.
